### PR TITLE
Rename `master_timeout` to `cluster_manager_timeout`

### PR DESCRIPTION
**Pull request by Yury-Fridlyand**
### Description
Signed-off-by: Yury Fridlyand [yuryf@bitquilltech.com](mailto:yuryf@bitquilltech.com)

* Rename `master_timeout` to `cluster_manager_timeout` everywhere it is used;
* Update doc links for `_cat/master` and `_cat/cluster_manager`;
* Add deprecation/introduction info for `CatMaster`/`CatClusterManager` APIs' and tests;
* Update json files with REST spec.

### CODE REVIEW

File: src/ApiGenerator/RestSpecification/Core/cat.allocation.json [hunk 1]
```diff
@@ -61,6 +61,10 @@
         "type":"time",
         "description":"Explicit operation timeout for connection to master node"
       },
+      "cluster_manager_timeout":{
```
Comment (acarbonetto) [inline src/ApiGenerator/RestSpecification/Core/cat.allocation.json:64]: do we want to remove "master_timeout" above?  I assumed "rename" means that you want to remove the old one...

File: src/ApiGenerator/RestSpecification/Core/cat.cluster_manager.json [hunk 2]
```diff
@@ -1,10 +1,14 @@
 {
   "cat.cluster_manager":{
     "documentation":{
-      "url":"https://opensearch.org/docs/latest/opensearch/rest-api/cat/cat-master/",
-      "description":"Returns information about the cluster-manager node."
+      "url":"https://opensearch.org/docs/2.0/opensearch/rest-api/cat/cat-cluster_manager/",
+      "description":"Returns information about the cluster_manager node. Introduced in OpenSearch 2.0 instead of cat.master."
```
Comment (acarbonetto) [inline src/ApiGenerator/RestSpecification/Core/cat.cluster_manager.json]: Fix: `cat.master` => `cat-master`

File: src/ApiGenerator/RestSpecification/Core/cat.cluster_manager.json [hunk 3]
```diff
@@ -1,10 +1,14 @@
...
     },
     "stability":"stable",
+    "visibility":"public",
```
Comment (acarbonetto) [inline src/ApiGenerator/RestSpecification/Core/cat.cluster_manager.json:8]: we want this?

File: src/ApiGenerator/RestSpecification/Core/cat.master.json [hunk 4]
```diff
@@ -1,8 +1,8 @@
 {
   "cat.master":{
     "documentation":{
-      "url":"https://opensearch.org/docs/latest/opensearch/rest-api/cat/cat-master/",
-      "description":"Returns information about the master node."
+      "url":"https://opensearch.org/docs/1.2/opensearch/rest-api/cat/cat-master/",
+      "description":"Returns information about the master node. Deprecated as of OpenSearch 2.0, use cat.cluster_manager instead."
```
Comment (acarbonetto) [inline src/ApiGenerator/RestSpecification/Core/cat.master.json]: Fix: `cat.cluster_manager` => `cat-cluster_manager`

File: tests/Tests/Cat/CatMaster/CatMasterApiTests.cs [hunk 5]
```diff
@@ -36,6 +36,7 @@
 namespace Tests.Cat.CatMaster
 {
 	[SkipVersion(">=2.0.0", "CatMaster API was renamed to CatClusterManager in 2.0.0 release")]
+	// TODO
```
Comment (acarbonetto) [inline tests/Tests/Cat/CatMaster/CatMasterApiTests.cs]: TODO?

**Review by acarbonetto — APPROVED** [review]
There's a TODO that needs fixing

File: src/ApiGenerator/RestSpecification/Core/cat.allocation.json (hunk 1 above)
Comment (Yury-Fridlyand) [reply src/ApiGenerator/RestSpecification/Core/cat.allocation.json:64]: No, it is kept for compatibility with OpenSearch 1.x.
Actually 2.0 still allows to use `master_timeout`, but it is in deprecated state.

File: tests/Tests/Cat/CatMaster/CatMasterApiTests.cs (hunk 5 above)
Comment (MaxKsyunz) [reply tests/Tests/Cat/CatMaster/CatMasterApiTests.cs]: Indeed. Why not create CatClasterManagerApiTests now?

Comment (Yury-Fridlyand) [reply tests/Tests/Cat/CatMaster/CatMasterApiTests.cs]: It was created in https://github.com/Bit-Quill/opensearch-net/pull/29/files#diff-e4a3470817fb4093031a3a1c2b530667b6a501d14243bf57f82f09ab07511cb5. I left TODO mark to check that nothing missed and forgot to delete it. Thank you for finding!

File: src/ApiGenerator/RestSpecification/Core/cat.cluster_manager.json (hunk 3 above)
Comment (Yury-Fridlyand) [reply src/ApiGenerator/RestSpecification/Core/cat.cluster_manager.json:8]: Copied from `cat.master.json`.

File: tests/Tests/Cat/CatMaster/CatMasterApiTests.cs (hunk 5 above)
Comment (Yury-Fridlyand) [reply tests/Tests/Cat/CatMaster/CatMasterApiTests.cs]: Fixed in 285868649.

File: src/ApiGenerator/RestSpecification/Core/cat.master.json (hunk 4 above)
Comment (Yury-Fridlyand) [reply src/ApiGenerator/RestSpecification/Core/cat.master.json]: Thank you, fixed in 285868649.

File: src/ApiGenerator/RestSpecification/Core/cat.cluster_manager.json (hunk 2 above)
Comment (Yury-Fridlyand) [reply src/ApiGenerator/RestSpecification/Core/cat.cluster_manager.json]: Thank you, fixed in 285868649

**Review by acarbonetto — APPROVED** [review]

**Pull request merged**